src/main/java/com/auctionapp/api/controller/AuthController.java

```diff
@@ -17,25 +19,30 @@
 public class AuthController {
 
     private final AuthService authService;
+    private final ValidationService validationService;
 
-    public AuthController(final AuthService authService) {
+    public AuthController(final AuthService authService, final ValidationService validationService) {
         this.authService = authService;
+        this.validationService = validationService;
     }
 
     @PostMapping("/register")
-    public ResponseEntity<String> register(@RequestBody final RegisterRequest registerRequest) {
-        String result = authService.validateRegisterRequest(registerRequest);
+    public ResponseEntity<ValidationRequest> register(@RequestBody final RegisterRequest registerRequest) {
+        String error = validationService.validateRegisterRequest(registerRequest);
 
-        if (result.isEmpty()) { 
-            result = authService.register(registerRequest); 
+        if (error.isEmpty()) {
+            ValidationRequest validationRequest = new ValidationRequest(authService.register(registerRequest), true);
+            return new ResponseEntity<>(validationRequest, HttpStatus.OK);
         }
-        return new ResponseEntity<>(result, HttpStatus.OK);
+
+        ValidationRequest validationRequest = new ValidationRequest(error, false);
```

Comment:
Not sure this naming is appropriate with how the class is used. This isn't a validation request but a validated response/validation response? Also, why is the login request treated differently?

Reply:
I have renamed it to validation response

register request with or without an error returns a string message, that I put in message field of validation response, while the login request if there is no error needs to return authentication response, that's why I wrote them differently